spark/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -303,96 +305,179 @@ public String toString() {
         }
       }
     }
+
+    private class EncryptedOutputFileFactory implements OutputFileFactory<EncryptedOutputFile> {
+      private final int partitionId;
+      private final long taskId;
+      private final long epochId;
+      // The purpose of this uuid is to be able to know from two paths that they were written by the same operation.
+      // That's useful, for example, if a Spark job dies and leaves files in the file system, you can identify them all
+      // with a recursive listing and grep.
+      private final String uuid = UUID.randomUUID().toString();
+      private int fileCount;
+
+      EncryptedOutputFileFactory(int partitionId, long taskId, long epochId) {
+        this.partitionId = partitionId;
+        this.taskId = taskId;
+        this.epochId = epochId;
+        this.fileCount = 0;
+      }
+
+      private synchronized String generateFilename() {
```

Comment:
I don't think this should be synchronized unless `fileCount` is volatile. This doesn't need to be synchronized anyway because each write is single-threaded. I would just remove this to make it a bit simpler.

Reply:
Agreed.